### ISSUE TITLE
Requests to https through proxy

### ISSUE THREAD

**Issue by Alek5andr** [axios/axios#3903]
#### Is your feature request related to a problem? Please describe.
Axios 0.21.1 does not cope with requests through proxy to secured URLs - HTTP code 501 is received.
I'm behind corporate network. When I try to launch the following code, the error 501 is received:
```
const axios = require("axios");

axios.get('https://jsonplaceholder.typicode.com/todos/12', {
    proxy: {
        protocol: 'http',
        host: 'corporate.proxy',
        port: 9000
    },
})
    .then(response => response.data)
    .then(data => console.log(data))
    .catch((error) => {
        const errorMessage = error.message;
        const responseData = error.response && error.response.data;
        console.log('error')
        console.log(`${errorMessage}${
            responseData
            && typeof responseData === 'object'
                ? `: ${JSON.stringify(responseData)}`
                : ''}`)
    });
```

Requests made to unsecured URLs - http protocol - with or without proxy settings succeed.

#### Describe the solution you'd like
Fix or implement please a possibility to perform requests to secured URLs - https protocol - through proxy.

#### Describe alternatives you've considered
Instead of using "axios" module, the module "axios-https-proxy-fix" solves the issue. However, it's some old fix. And most probably does not update to new features that original Axios may bring. Definitely I would like to stick to primary.

#### Additional context
No solution is found around the original source.

**Comment by daniel-julius-sam** [comment]
When I looked in the docs it says you have to give `protocol: 'https'` for the proxy. Here's a link to the docs. [https://axios-http.com/docs/req_config](url)

**Comment by lolyinseo** [comment]
This issue about  4 years (https://github.com/axios/axios/issues/925)
They can't fix it, use another lib

**Comment by mbargiel** [comment]
@jasonsaayman I'm coming up with a PR to fix this soonish (1-2 weeks). At first I used `https-proxy-agent` but there are too many issues with it (and it's unmaintained) so there would be a couple of ugly workarounds in the axios code, as well as some uncovered use cases. However I discovered `hpagent`, which looks like a more promising contender. I just need to test it some more and get one issue in it fixed before I can open an axios PR to use it and fix all the issues related to "proxying to HTTPS not working".
